pkg/controllers/state/statenode.go

```diff
@@ -517,3 +518,31 @@ func RequireNoScheduleTaint(ctx context.Context, kubeClient client.Client, addTa
 	}
 	return multiErr
 }
+
+// ClearNodeClaimsCondition will remove the conditionType from the NodeClaim status of the provided statenodes
+func ClearNodeClaimsCondition(ctx context.Context, kubeClient client.Client, conditionType string, nodes ...*StateNode) (err error, requeue bool) {
+	return multierr.Combine(lo.Map(nodes, func(s *StateNode, _ int) error {
+		if !s.Initialized() || s.NodeClaim == nil {
+			return nil
+		}
+		nodeClaim := &v1.NodeClaim{}
+		err := kubeClient.Get(ctx, client.ObjectKeyFromObject(s.NodeClaim), nodeClaim)
+		if err != nil {
+			return err
+		}
+		if !nodeClaim.StatusConditions().IsTrue(conditionType) {
+			return nil
+		}
+		stored := nodeClaim.DeepCopy()
+		_ = nodeClaim.StatusConditions().Clear(conditionType)
+
+		if err := kubeClient.Status().Patch(ctx, nodeClaim, client.MergeFromWithOptions(stored, client.MergeFromWithOptimisticLock{})); err != nil {
+			if errors.IsConflict(err) {
+				requeue = true
+				return nil
+			}
```

Comment:
I think personally I'd rather we not include reconcile.Result as a return value and then do a `errors.IsConflict()` in the call in the disruption controller, so that we leave the error handling up to the controller. Should make the code easier too